src/worker/send_notification.cr

```diff
@@ -0,0 +1,59 @@
+require "sidekiq"
+
+require "../application"
+require "../bot"
+require "../bad_dragon/entities/toy"
+require "../errors"
+require "../repository/lock"
+require "../repository/watch_notifications"
+
+module Worker
+  class SendNotification
+    include Sidekiq::Worker
+
+    sidekiq_options do |job|
+      job.queue = "send_notification"
+    end
+
+    def perform(watcher : Int64, toy : BadDragon::Entities::Toy)
+      loop do
+        begin
+          Repository::Lock.lock("toys_#{watcher}") do
+            Application.logger.info "[notify] notifying #{watcher} for #{toy.sku}/#{toy.size}"
+            send_messages(watcher, toy)
+            Repository::WatchNotifications.add(watcher, toy)
+          end
+          break
+        rescue e : Errors::LockUnavailableError
+          sleep 5 * rand
```

Comment:
Sidekiq retries are too slow for that, so I retry it here after sleeping some time (max 5 seconds)